src/WebJobs.Script.WebHost/Metrics/LinuxContainerLegionMetricsPublisher.cs

```diff
@@ -0,0 +1,251 @@
+
+        private TimeSpan _metricPublishInterval;
+        private Process _process;
+        private Timer _processMonitorTimer;
+        private Timer _metricsPublisherTimer;
+        private bool _initialized = false;
+
+        public LinuxContainerLegionMetricsPublisher(IEnvironment environment, IOptionsMonitor<StandbyOptions> standbyOptions, ILogger<LinuxContainerLegionMetricsPublisher> logger, IFileSystem fileSystem, ILinuxConsumptionMetricsTracker metricsTracker, IScriptHostManager scriptHostManager, int? metricsPublishIntervalMS = null)
+        {
+            _standbyOptions = standbyOptions ?? throw new ArgumentNullException(nameof(standbyOptions));
+            _environment = environment ?? throw new ArgumentNullException(nameof(environment));
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            _metricsTracker = metricsTracker ?? throw new ArgumentNullException(nameof(metricsTracker));
+            if (scriptHostManager == null || !Utility.TryGetHostService<IMetricsLogger>(scriptHostManager, out _metricsLogger))
+            {
+                throw new InvalidOperationException("Unable to get IMetricsLogger Service.");
+            }
+            _containerName = _environment.GetEnvironmentVariable(EnvironmentSettingNames.ContainerName);
+            _metricPublishInterval = TimeSpan.FromMilliseconds(metricsPublishIntervalMS ?? 30 * 1000);
+
+            // Default this to 15 minutes worth of files
+            int maxFileCount = 15 * (int)Math.Ceiling(1.0 * 60 / _metricPublishInterval.TotalSeconds);
+            string metricsFilePath = _environment.GetEnvironmentVariable(EnvironmentSettingNames.FunctionsMetricsPublishPath);
+
+            _metricsFileManager = new LegionMetricsFileManager(metricsFilePath, fileSystem, logger, maxFileCount);
+
+            _processMonitorTimer = new Timer(OnProcessMonitorTimer, null, Timeout.Infinite, Timeout.Infinite);
+            _metricsPublisherTimer = new Timer(OnFunctionMetricsPublishTimer, null, Timeout.Infinite, Timeout.Infinite);
+
+            _metricsTracker.OnDiagnosticEvent += OnMetricsDiagnosticEvent;
+
+            if (_standbyOptions.CurrentValue.InStandbyMode)
+            {
+                _standbyOptionsOnChangeSubscription = _standbyOptions.OnChange(o => OnStandbyOptionsChange());
+            }
+            else
+            {
+                Start();
+            }
+        }
+
+        public void Initialize()
+        {
+            _process = Process.GetCurrentProcess();
+            _initialized = true;
+        }
+
+        public void Start()
+        {
+            Initialize();
+
+            // start the timers by setting the due time
+            SetTimerInterval(_processMonitorTimer, _timerStartDelay);
+            SetTimerInterval(_metricsPublisherTimer, _metricPublishInterval);
+
+            _logger.LogInformation(string.Format("Starting metrics publisher for container : {0}.", _containerName));
```

Comment:
Don't use string format or interpolation in log statements:

```suggestion
            _logger.LogInformation("Starting metrics publisher for container: '{containerName}'.", _containerName);
```

Reply:
Done